common/rpc/encryption/tlsFactory.go

```diff
@@ -60,10 +63,29 @@ type (
 		GetCertProvider(hostName string) (CertProvider, error)
 	}
 
+	CertExpirationData struct {
+		Thumbprint [16]byte
+		IsCA       bool
+		DNSNames   []string
+		Expiration time.Time
+	}
+
+	CertExpirationMap map[[16]byte]CertExpirationData
```

Comment:
[Nit] mapping Thumbprint to 16[byte] and keeping map[Thumbprint]CertExpirationData instead of replacing with CertExpirationMap feels it might be easier to read